src/codegen/src/main/scala/WrapperGenerator.scala

```diff
@@ -59,16 +61,18 @@ abstract class WrapperGenerator {
              generateTransformerTestWrapper(t, className, qualifiedClassName))
           case e: Estimator[_] =>
             val sc = iterate[Class[_]](myClass)(_.getSuperclass)
-                     .find(c => Seq("Estimator", "Predictor").contains(c.getSuperclass.getSimpleName))
+                     .find(c => Seq("Estimator", "ProbabilisticClassifier", "Predictor")
+                       .contains(c.getSuperclass.getSimpleName))
                      .get
             val typeArgs = sc.getGenericSuperclass.asInstanceOf[ParameterizedType]
               .getActualTypeArguments
             val getModelFromGenericType = (modelType: Type) => {
               val modelClass = modelType.getTypeName.split("<").head
-              (modelClass.split("\\.").last, modelClass)
+              (modelWrapperName(myClass, modelClass.split("\\.").last), modelClass)
             }
             val (modelClass, modelQualifiedClass) = sc.getSuperclass.getSimpleName match {
               case "Estimator" => getModelFromGenericType(typeArgs.head)
+              case "ProbabilisticClassifier" => getModelFromGenericType(typeArgs(2))
```

Comment:
Why do we need to specifically code against this API here? seems like this muddies the abstraction and is not needed becase a Probabbalistic Classifier is a predictor

Reply:
I needed probabilistic classifier to get the model type, otherwise in the autogenerated lightgbm wrappers the model was "M" instead of lightgbmclassificationmodel